src/structure/immutable/deepEqual.js

```diff
@@ -1,20 +1,22 @@
 import { Iterable } from 'immutable'
-import every from '../../util/every'
 
-const deepEqualValues = (a, b) => {
-  if (a === undefined && b === '') return true
-  if (a === '' && b === undefined) return true
-  if (Iterable.isIterable(a)) {
-    if (Iterable.isIterable(b)) {
-      return a.count() === b.count() &&
-        a.every((value, key) => deepEqualValues(value, b.get(key)))
-    }
-    return false
-  }
-  if (!a || !b || typeof a != 'object' && typeof b != 'object') {
-    return a === b
+import isEqualWith from 'lodash.isEqualWith'
+
+const customizer = (obj, other) => {
+  if (obj === undefined && other === '') return true
+  if (obj === '' && other === undefined) return true
+
+  if (Iterable.isIterable(obj) && Iterable.isIterable(other)) {
+    return obj.count() === other.count() && obj.every((value, key) => {
+      return isEqualWith(value, other.get(key), customizer)
+    })
   }
-  return every(a, (value, key) => deepEqualValues(value, b[key]))
+
+  return void 0
+}
+
+const deepEqualValues = (a, b) => {
+  return isEqualWith(a, b, customizer)
 }
```

Comment:
Arrow function shave implicit return so if it's within your style preference you could do:
```js
const deepEqualValues = (a, b) => isEqualWith(a, b, customizer)
```